src/main/haskell/kore/src/Kore/Step/Condition/Evaluator.hs

```diff
@@ -128,13 +129,18 @@ refutePredicate
 refutePredicate korePredicate =
     case isMetaOrObject (Proxy :: Proxy level) of
         IsMeta   -> return Nothing
-        IsObject ->
-            SMT.inNewScope $ runMaybeT $ do
-                smtPredicate <- goTranslatePredicate korePredicate
-                SMT.assert smtPredicate
-                result <- SMT.check
-                case result of
-                    Unsat -> return False
+        IsObject -> SMT.inNewScope $ runMaybeT $ do
```

Comment:
What is this about?

Reply:
We now check if predicates are always-true (to delete the side condition) or always-false (to prune the branch).

Actually, right, I should rename that function.